File: python/bob/bob.py
import re

def hey(what):
    what = what.strip()
    with_force = (re.search("[A-Za-z]",what) and what.upper() == what)
    if re.search("\?$", what) and not with_force:
        return "Sure."
    elif "" == what:
        return "Fine. Be that way!"
    elif with_force:
        return "Whoa, chill out!"
    else:
        return "Whatever."

File: ecmascript/bob/bob.js
class Bob {
  hey(message) {
    if (message.trim().length == 0)
      return 'Fine. Be that way!'
    else if (message.toUpperCase() == message && /[A-Za-z]/.test(message))
      return 'Whoa, chill out!'
    else if (message.slice(-1) == '?')
      return 'Sure.'
    else
      return 'Whatever.'
  }
}

export default Bob;

